docs/standard/io/file-path-formats.md

```diff
@@ -0,0 +1,170 @@
+
+- For a single period, the current segment is removed, since it refers to the current directory).
+
+- For a double period, the current segment and the parent segment are removed, since the double period refers to the parent directory.
+
+   Parent directories are only removed if they aren't past the root of the path. The root of the path depends on the type of path. It is the drive (`C:\`) for DOS paths, the server/share for UNCs (`\\Server\Share`), and the device path prefix for device paths (`\\?\` or `\\.\`).
+
+### Trimming characters
+
+Along with the runs of separators and relative segments removed earlier, some additional characters are removed during normalization:
+
+- If a segment ends in a single period, that period is removed. (A segment of a single or double period is normalized in the previous step. A segment of three or more periods is not normalized and is actually a valid file/directory name.
+
+- If the path doesn't end in a separator, all trailing periods and spaces (U+0020) are removed. If the last segment is simply a single or double period, it falls under the relative components rule above. 
+
+   This rule mans that you can create a directory name with a trailing space by adding a trailing separator after the space. Trailing spaces can make it difficult to access a directory.
+
+## Skipping normalization
+
+Normally, any path passed to a Windows API is (effectively) passed to the [GetFullPathName function](https://msdn.microsoft.com/library/windows/desktop/aa364963(v=vs.85).aspx) and normalized. There is one important exception: a device path that begins with a question mark instead of a period. Unless the path starts exactly with `\\?\` (note the use of the canonical backslash), it is normalized.
+
+Why would you want to skip normalization? There are three major reasons:
+
+1. To get access to paths that are normally unavailable but are legal. A file or directory called `hidden.`, for example, is impossible to access in any other way. 
+
+1. To improve performance by skipping normalization if you've already normalized.
+
+1. To skip the `MAX_PATH` check for path length to allowing for paths that are greater than 259 characters. Most APIs allow this, with some notable exceptions, such as the [GetCurrentDirectory](https://msdn.microsoft.com/library/windows/desktop/aa364934(v=vs.85).aspx) and [SetCurrentDirectory](https://msdn.microsoft.com/library/windows/desktop/aa365530(v=vs.85).aspx) functions.
+
+Skipping normalization and max path checks is the only difference between the two device path syntaxes; they are otherwise identical. Be carefully with skipping normalization, since you can easily create paths that are difficult for "normal" applications to deal with.
+
+Paths that start with `\\?\` are still normalized if you explicitly pass them to the [GetFullPathName function](https://msdn.microsoft.com/library/windows/desktop/aa364963(v=vs.85).aspx).
```

Comment:
However .NET does skip normalizing them in `Path.GetFullPath()`.  The reason is that we implicitly call `Path.GetFullPath()` from almost all of our APIs that take paths. We need to do so to provide implicit long path support and we rely on normalized paths for simpler, more robust logic (much as Windows does internally).